tableauserverclient/server/server.py

```diff
@@ -47,6 +58,29 @@ def _set_auth(self, site_id, user_id, auth_token):
         self._user_id = user_id
         self._auth_token = auth_token
 
+    def _get_legacy_version(self):
+        response = self._session.get(self.server_address + "/auth?format=xml")
+        info_xml = ET.fromstring(response.content)
+        prod_version = info_xml.find('.//product_version').text
+        version = _PRODUCT_TO_REST_VERSION.get(prod_version, '2.1')  # 2.1
+        return version
+
+    def _determine_highest_version(self):
+        try:
+            old_version = self.version
+            self.version = "2.4"
+            version = self.server_info.get().rest_api_version
+        except ServerInfoEndpointNotFoundError:
+            version = self._get_legacy_version()
+
+        finally:
```

Comment:
I currently have it this way so that this function simply *retrieves* the highest supported version, and the caller is responsible for setting it wherever it wants to be set -- this lets us reuse the function elsewhere (I don't know a use case yet, but I don't think the premature refactor makes it harder to follow).

I'd prefer the separation, if there's a strong objection let me know and I can combine it.

I'm pro-flag that does this automatically!!! I'll let Lee weigh in and can take a look